### PR TITLE
feat(game): the GM writes back to Experience packages through declared verbs (#5798)

### LINKED ISSUE #5798: A way for the GM to write back to an Experience package
# A way for the GM to write back to an Experience package

## What's missing, in one paragraph

An Experience package can *read* the chat: the host hands it props every turn, and it can
subscribe to a small set of engine-defined events. There is no lane in the other direction. When
the GM narrates "the storm rolls in" or "the innkeeper waves off your coin and gives you the room",
the package never hears about it, so the sky stays fair and the purse stays full. Everything the
narrator does to the world is invisible to the surface the player is standing on, and the only
thing keeping the two in step is that the package writes a few words into the prompt header and
hopes.

What this asks for is the reverse lane: **a package-addressed, structured output channel** — a
small closed vocabulary of named calls, with primitive arguments, validated by the server before
delivery, that reaches the package the way host events already do.

## Why this shape and not a free-form one

The founding rule of Pixelforge is *the model decides what happened; the algorithm decides how it
looks.* A free-form channel would hand the model geometry authority and re-open every seeding and
determinism guarantee the package is built on. A closed enum of verb names with typed arguments
does not: every call is a selection from a list the package published, and the package can refuse
one that does not fit the world it actually compiled.

## The delivery mechanism already exists

The Capability API 1.12 event bus is the pattern. The host dispatches a browser event —
`CAPABILITY_CLIENT_EVENT`, `"marinara-capability-server-event"`
(`packages/client/src/lib/capability-client-events.ts:1`) — and a package's element filters it on
package id and chat id. Pixelforge binds that listener once
(`packages/pixelforge/src/90-element.js:819`, in the Pixelforge repo) and routes it into a handler
that already consumes three event types (`packages/pixelforge/src/50-spatial.js:139`). What it
cannot do today is carry anything the *model* said: the type strings are engine-defined, so there
is no surface a real writer could sit on. That is the verified reason Pixelforge 0.14 ships a fully
working weather override with a browser console as its only writer.

So the ask is not a new transport. It is **the same event bus, with a payload the model can
produce** — concretely, a schema-validated custom tool call whose name is drawn from a closed enum
the package declares.

**Version:** this wants the next Capability API point release. The registry is
`packages/shared/src/schemas/capability-package.schema.ts` — `supportedCapabilityApi`. Two
observations, because the trees disagree: the review worktree at merge `34442e26d` reads
`{major: 1, minor: 13}` (line 184), and `feat/mari-write-readback` at `af2dc003a` reads
`{major: 1, minor: 15}` (line 196). Whichever is current when this lands, the point release above
it is the ask. Pixelforge's own declaration is `{major: 1, minor: 10}` and does not move this.

## The four vocabularies that want the channel

They are listed together because they are one channel and one validation story, not because they
have to ship together. Any subset is a useful first cut.

### 1. World flags — six entries

`closed:<zone>`, `damaged:<zone>`, `shelter:<zone>`, `hold:<npc>@<zone>`, `gone:<npc>`,
`hail:<npc>`.

Write-only, world-mutating. These are the ones that make the narrator's fire leave a burnt barn.
One of them already has a consumer built and waiting: `_hold` is read by the schedule resolver
(`packages/pixelforge/src/30-sim.js:583` — "`_hold` reserves a GM override seam") and is the seam
that keeps a held NPC where the story put them.

**These need a decision on the package side before they can ship at all** — see "What this is
not" below. The channel is not what blocks them.

### 2. Weather — `weather:<word>[:untilDay]`, with an optional intensity

The word comes from a five-entry enum: `fair`, `overcast`, `rain`, `storm`, `snow`. `rain` and
`snow` additionally take `intensity: "light" | "heavy"`, so a GM can summon *heavy rain* and a bare
`rain` gets the intensity the day's climate would have rolled anyway.

**This one is read-complete today.** Pixelforge 0.14 derives the sky from the world's own climate
and calendar, and reads an override out of chat metadata at the key `pixelforgeWeather` — folding
it defensively, never writing it back, and re-placing the town on the next props delivery when the
key changes. A writer is the only missing half.

The landing is the metadata key, not a new store: the writer patches `pixelforgeWeather` (per-key
shallow merge), stamping `sinceDay` = the live sim day at write time. The package observes the
change, re-folds it, and calls its own schedule resolver — the town answers a mid-day override with
no contact with the prompt envelope at all.

Three duties the writer inherits, stated so they are priced rather than discovered:

- **An expired row is inert.** Once `untilDay` is past, every reader treats the row as absent. The
  next write replaces it or nulls it; nothing garbage-collects it.
- **It does not rewind with the story.** The override is chat-scoped configuration, like the world
  brief and the content pack. It is a *day-range* predicate: rewinding back INTO
  `[sinceDay, untilDay]` re-arms it, which is correct — that is the sky that day already had.
  Rewinding to before it was set restores the derived sky. A GM who wants a different old sky
  writes again.
- **A vanished row means "no override", never corruption.** See the durability note below.

### 3. Economy verbs — four entries

`rentBerth`, `grant`, `take`, `award` — or a narrower subset.

These call straight through to mutators the package already ships and already fences, so they need
**no new package-side plumbing to execute**. What they need is the schema gate in front of them and
new call-site logic — and that second half is real work, for a reason worth naming up front rather
than discovering during implementation.

**The proximity asymmetry, and the decision.** The shipped verb *functions* (as distinct from the
raw mutators underneath them) assume a player-proximity precondition the GM does not have.
`berthOffer` resolves a keeper standing next to the player and refuses when there is none
(`packages/pixelforge/src/59-economy.js:693`); `rentBerth` re-reads that offer before it moves a
single field (`:729`). A GM-narrated purchase satisfies no such precondition by construction — the
innkeeper in the prose may be three zones away.

**Decided, so it is in the estimate:** GM verbs carry **validated target arguments against explicit
allowlists** and are executed as a **second, GM-shaped implementation** over the same gen-fenced
mutators. The GM says *which* keeper, *which* zone, *which* item, and the package validates each
against the live compiled world before executing. This is more work than "expose the existing
functions", and pretending otherwise is how the spec freezes and then grows.

### 4. One tuning row — `fishBite:<word>:<multiplier>`

Pixelforge's fishing bite rate is conditioned on the sky: wet and grey days bite better. The
multipliers live in one tuning table keyed by weather word. Worth stating precisely, because it
is the kind of gap that becomes a bug: the validation enum has FIVE words and the table has FOUR
keys. Fair weather is deliberately absent — the lookup falls back to a multiplier of one — so a
write validated against the five-word enum must not be allowed to CREATE a `fair` key that the
fallback would then start reading. Validate the word against the five; accept a write only for
the four the table names.

**Honest note, because a reader will ask:** the weather override *already* tweaks bites
implicitly — summoning rain is summoning the multiplier. The dedicated row exists to tune the
multiplier itself, not to get faster bites. It is the smallest of the four and the easiest to cut.

**Adjacent, named and not designed:** a GM time-skip verb (`advanceDays:<n>`) is a natural
candidate for this channel. Pixelforge's own time verbs are day-grain and its seasons are months
long, so a GM who wants to show the player a winter has no way to get there. Maintainer's option;
nothing here designs it.

## The validation split

Two gates, and they answer different questions:

- **The engine validates shape and enum membership before delivery.** Is this a name in the
  declared vocabulary, are the arguments the declared primitive types.
- **The package re-validates at the point of use against the live compiled world.** A flag naming a
  zone id that does not exist in *this* world is well-formed and still wrong.

Neither substitutes for the other. This is the same seal-time discipline the package already
applies to LLM-authored world briefs: validate the shape once, then re-check the meaning against
whatever is actually standing.

## Size is not a constraint here

Said explicitly so the engine side does not over-design for a budget that does not exist: a six-to
-ten-entry closed enum with primitive arguments is nowhere near the 262,144-character per-row
snapshot wall. There is no need for compression, chunking, or a separate store.

## One durability note the engine side should know

`pixelforgeWeather` inherits the **#5076-class whole-blob-erase hazard** (the issue itself shipped; the class is what the package's own save code names): roughly forty engine call sites
still use the unqueued whole-blob `updateMetadata`, any of which can silently erase a package's
metadata key between turns. Pixelforge already carries a self-heal for two of its keys for exactly
this reason (`packages/pixelforge/src/60-save.js:1493`, `ensurePresent`).

The override key gets **no self-heal**, deliberately — there is nothing to re-derive it from. So
the writer must treat a vanished row as **"no override"**, never as corruption, and re-write rather
than attempt repair. If #5076 is fixed before this ships, this note retires; until then it is a
behaviour the writer has to have.

## What this is explicitly NOT

**It is not a request for the engine to solve the world-flag persistence question.** The flags in
vocabulary 1 mutate a world that Pixelforge rebuilds from its seed on every load, so without a
persistence contract every consequence dies at the next reload. That decision — a sibling
world-state block, host-authoritative re-read, or session-only — is a package-side call, scoped on
the Pixelforge roadmap, and has to be made before those flags ship regardless of what this channel
looks like. Weather, notably, does not have this problem: the override lives in chat metadata, so
the metadata key *is* the persistence.

**It is not a free-form state patch.** If the answer is "use the tracker/state-patch surface", that
is worth saying — but the closed enum is the requirement, not the transport.

## Where a reviewer would start

- `packages/shared/src/schemas/capability-package.schema.ts` — `supportedCapabilityApi`, the
  version registry.
- `packages/client/src/lib/capability-client-events.ts` — the existing package-addressed delivery
  this extends.
- `packages/shared/src/schemas/custom-tool.schema.ts` — the closest existing shape for "a validated
  call the model can make".

Filed on behalf of the Pixelforge Experience package; the four vocabularies above are its own, and
the weather half is built and read-complete as of Pixelforge 0.14.0.

### PR DESCRIPTION
**Pull request by kolacheee**
> [!WARNING]
> **Stack note — this PR is stacked on the GM verb schema PR. Merge that one first.** Until it merges, the diff below also contains its five commits (the shared schema file, its docs section, its regression and its CHANGELOG line). Review those there, not here; once it lands, this diff shrinks to the eleven commits that add the runtime.

> [!IMPORTANT]
> Contributions target `staging`. Only `SpicyMarinara` may promote this repository's `staging` branch or a same-repository `hotfix/*` branch to `main`.
> Outside and first-time contributors also require an approving review from `SpicyMarinara`.

## Linked issue

Closes #5798

## Why this change

An Experience package can read a chat and cannot write to it. The host hands the package props every turn and dispatches a small set of engine-defined events at it, but nothing carries anything the *model* said. So when the GM narrates the storm rolling in, the package never hears; the sky stays fair, the town keeps its fair-weather schedule, and the only thing holding the story and the world together is that the package writes a few words into the prompt header and hopes.

The request that opened this names four vocabularies that want the lane: world flags (a burnt barn stays burnt, a held NPC stays put), weather, economy verbs, and one fishing tuning row. They are one channel and one validation story, not one release — any subset is a useful first cut.

**Weather is first because it is the one that is already read-complete.** Pixelforge derives the sky from the world's own climate and calendar and *already* reads an override out of chat metadata, folds it defensively, and re-places the town on the next props delivery when the key changes. It has shipped that reader with a browser console as its only writer. Every piece of that feature exists except the half that lets the narrator use it. This PR is that half, built generically: the Engine does not know what weather is, only that a package declared an action, that the model used it, and that the values check out.

The second reason to do it now is that the package needing it cannot run server code at all. Pixelforge declares only agent and client entrypoints, so it can never register a server-side command handler — the channel has to be declaration-driven with an Engine-generic executor, or it does not reach the package that asked for it.

## What changed

**One table resolution per turn, threaded to both ends.** The chat's Experience id resolves to an installed package, its action table is read and validated once, and that one resolution is handed to both the prompt render and the post-save scan. Two separate resolutions could disagree — a package updated mid-turn, a table that stops verifying — and then the prompt would advertise an action the parser no longer matches, leaving a raw bracket tag sitting in the player's prose. The resolution is lazy, so a chat with no Experience package pays nothing.

**A gated reader, `gmVerbTableSource(packageId)`.** The verified-file reader is module-private and there is no exported by-id package accessor, so the runtime gets exactly one narrow export that returns verified bytes or nothing and never hands back an installed-package object. Its gate chain is: package readiness (not servability — the servable path falls back to the *previous* version's manifest for a package awaiting restart, which would serve a vocabulary the running Engine no longer matches), then the file actually being a declared asset, then the `chat-write` permission, then a 64 KB check against the *declared* byte count before any read, then the verified read. Failure tiers are the function's contract and the turn survives all of them: not installed, not ready or no table declared are silent; a packaging mistake warns; a hash or tamper failure logs an error naming tampering.

**The tag grammar is shared, not re-written.** The conversation-command tag pattern is exported as a source string and the new scan uses it against a different vocabulary. An action the prompt advertises but the parser cannot match would leave a raw tag in the player's prose, and two copies of a regex is how that happens.

**A new engine-generic executor** resolves, renders, parses, validates and executes, with no package server code anywhere in the path.

**Render, parse and strip on the game turn.** Action lines render inside the GM format reminder's `COMMANDS:` block through a new parameter beside the existing experience systems list — not the system message, which is rebuilt wholesale later and would overwrite anything spliced into it. The scan is its own game-mode-only path: the conversation-commands flag gates that entire surface, so flipping it here would arm every registered conversation command on every game turn, and it stays untouched. The scan runs before the content-replacement frame, because anything that must reach the client has to be in the response by then. **A tag is removed from the saved text on the name match, never on validation success** — a command the model got slightly wrong must not show the player machinery.

**Its own execution site.** Not the conversation-command block: that one brackets its work in frames the game client does not consume, and never runs in Game Mode anyway. Nothing executes on an aborted turn.

### The executor's write discipline

- **`patchMetadata` and nothing else.** Never `updateMetadata`, never the capability persistence host. The doc comment states the rule at the function.
- **Absolute writes.** An action's validated arguments are written wholesale under its key. The merge is a top-level shallow spread, so the write replaces the whole value. *That* — not the claim record — is why a duplicate apply is harmless, and it is equally why a relative action ("add 5 gold") cannot be expressed here by construction. A duplicate identical write moves no ordinal and costs nothing.
- **`touchUpdatedAt: false`**, so a weather change does not reorder the user's chat list.
- **Write, then claim.** Each executed action claims `gmVerb:<name>` on the saved swipe with a truthy object value (`{verb, args, at}`) — a scalar would leave the storage guard a permanent no-op. The claim is skipped, never the write, when the saved message id is empty. **The claim buys provenance, not dedupe**, and the code says so where it is written.
- **Refusals are per action, per turn and per call.** A reserved name or a foreign key drops that action and the rest of the table still runs; an invalid payload drops that call; one action throwing never costs another its effect.
- **One `metadata_patch` frame** is emitted if any write committed, flagged the instant the row commits rather than after the claim — a committed write the client is never told to refetch would leave the package showing a stale world. The payload itself is inert: the client's handler invalidates the chat detail and the active lorebook and ignores the data. The props re-delivery after that refetch *is* the delivery mechanism for a written value; there is no second event carrying it.

### The live-delivery half, and its honesty

An action that names no key is delivered to the package instead of written: one `gm_verb` frame carrying an explicit `packageId` plus the chat, message and swipe it came from, dispatched into the shipped capability client-event bridge. The address is explicit because the executor already holds the resolved package id — the client is not asked to infer it from a convention field inside the payload the way the existing turn-game bridge does. The case lands in **both** SSE switches in the same change, including the agents-retry switch where GM actions do not run today, so the next wiring does not have to rediscover that the two are twins.

What this half honestly is:

- **The channel has no memory.** One frame, one synchronous browser dispatch. No queue, no replay, no acknowledgement. An effect is lost — silently — on an aborted turn (the stream is already gone), a tab closed or reloaded mid-stream, a dispatch that arrives before the package's element has mounted for the first time in that page's life, a chat the player has switched away from, and the package's own loading gate. Nothing re-delivers it.
- **One warning is the only trace.** An unwritable reply is warned about; ordinary backpressure is not, because a queued frame is not a lost one, and folding the two would make the warning meaningless.
- **The claim cannot buy at-most-once here even in principle** — it is written after the frame, and the client may never have received the frame.
- **The regeneration nuance, stated plainly because it shapes what may ship on this half.** Regenerating a turn mints a fresh swipe, and a new swipe's extras are rebuilt from a fixed allowlist that does not carry these claims forward. So a *relative* live-delivered action would apply once per swipe generated, and no key on the message can stop it: the key guards redelivery of one generation, and redelivery is the one thing a channel with no replay cannot do. This is why relative shapes are refused at declaration and why the first action on this half is one whose package-side write is absolute in every field.
- **The two halves genuinely differ on rewind.** A written chat setting does not rewind with the story — it survives swiping away, editing or deleting the turn. A live-delivered effect applied into a package's own state does revert with a rewind, because the package rebuilds that state from its save.
- **Ordering on a turn that carries both.** The live dispatch is synchronous and reaches the package *before* the written value's refetch lands. A handler must not read a same-turn written value and expect the new one. The docs say this rather than leaving it to be found.
- **Refusal asymmetry.** The Engine can only check shape; an NPC name cannot be enumerated at declaration time, because the world is compiled per chat. So the package's own re-validation is advisory for a written value (the row is already committed) and **binding** for a live-delivered one (nothing was committed Engine-side). That asymmetry is a feature and the docs state it.

### The `chat-write` gate, and what it widens

Reading a package's table requires the package to declare `chat-write`, and the permission check sits ahead of the read so an unpermitted package's bytes are never loaded — and behind the declaration test, so a package with no table stays silent while one that ships a table and forgot the permission is told.

**Stated up front rather than discovered:** this changes what that permission means. Today `chat-write` means "this package's own client code writes chat data". After this PR it also means "the Engine will execute writes on this package's behalf, from model output". Permissions are shown before install and nothing re-consents on update, so a user who granted the old meaning receives the new one without being asked again. The separate issue about capability permissions that are declared and enforced nowhere (#5899) carries this note; the first counter-example to that issue is this gate.

Practically the blast radius today is zero: no released package ships an action table at all, so no install changes behavior until one does.

### Capability API 1.16

The bump waited for this PR on purpose. The compatibility check compares a package's declared version against that constant alone, so bumping it advertises the seam whether or not the runtime exists. It exists now. It carries its own line in the version ladder, and the lifecycle regression pins the constant and its compatibility message in four places so the bump moves them with it.

### Teaching the model the vocabulary it is allowed to use

The reminder line originally rendered a name, a description and one example, so an action's closed vocabulary reached the GM only if the package happened to spell it inside a 200-character description. A GM told to set the weather writes "sunny" for "fair"; the validator refuses a word it was never shown; and **the refusal is invisible**, because the tag is stripped on the name match — the narration reads clean and the world simply never changed.

The line now carries the two payload slots the sibling conversation-command renderer already uses, and the first one is derived from the same parsed table the validator reads: every argument in declaration order, optional ones marked outside the JSON string, a closed list rendered as the full alternation the built-in commands already use, an open string shown with its cap, numbers and booleans unquoted because the validator refuses `"3"` for a number rather than coercing it. Deriving it here instead of from prose is what stops prompt and validator drifting: a description can no longer promise a value the validator refuses.

### One prompt hazard, named

The GM format reminder already renders a built-in `[reputation: npc= action=]` line on every Game Mode turn, consumed by the Engine's own relative reputation service against records Pixelforge never populates. The reserved-name rule correctly makes `reputation` an illegal package action name — but a package's own standing action will sit one line away from a do-nothing Engine stance verb in the same block, and a model handed two overlapping stance verbs will sometimes reach for the wrong one. The name rule cannot fix that. Suppressing the Engine line on chats whose package declares a standing action is a candidate follow-up, not part of this PR.

### Also in this PR

- The retry SSE switch gains `metadata_patch`, which is load-bearing rather than tidy: the retry route has been emitting that frame with no case to receive it, so a metadata write on a retried turn never reached the package until the chat was reopened.
- A misleading comment in the conversation-actions regression is fixed. It claimed "regeneration cannot double-send" four lines above an assertion that the handler ran twice. The dispatch key is an in-flight lock released in a `finally`, not a ledger; what the second call refuses is the claim, after the handler has already run. That comment is where a false safety claim in the original design notes came from.

## Known shortcuts

Named here so review does not have to find them:

1. **One instance per action name per message.** Correct for the two proving actions; a real ceiling for any future repeatable live-delivered action, where a repeat is meaningful prose ("two coins, then two more"). A payload-hash dedupe behind a `repeatable` flag is sketched, not built.
2. **Flat, single-line JSON payloads only.** Argument values are string, number or boolean; there is no nesting.
3. **A written value survives swiping away, editing or deleting the turn, and nothing reverts it.** The visible symptom is that the last swipe *generated* wins, not the last *displayed* — the prose on screen and the world can disagree within a session with nothing reconciling them. A live-delivered effect does revert with a rewind; the halves genuinely differ here.
4. **The claim buys provenance, not dedupe** — and on the live-delivery half it cannot buy at-most-once even in principle. Safety comes entirely from the write being absolute.
5. **A package's semantic refusal of a written value is advisory**; the row is already committed. A refusal of a live-delivered action is binding.
6. **`Math.random()` stays unseeded everywhere.** There are three separate unseeded implementations in the tree; this PR neither uses nor fixes them.
7. **After an update that leaves a package awaiting restart, its actions stop resolving until the restart** — a log line only, no user-facing notice. No shipped package can trip this today.
8. **A GM turn that is nothing but an action tag now lands rather than being discarded** — a review round pulled the narrow fix in: the empty-response guard's hidden-anchor gate now counts parsed GM verb calls, so a tags-only turn saves its hidden anchor, executes its validated writes, and shows the player a silent turn instead of an error; the game surface skips the hidden anchor when picking the narration to display, so the previous turn's text stays on screen. #5902 remains open for the generic anchor surface (the gate still cannot see command shapes it was never taught), and two follow-ups were filed from the review: #5926 (a command-only regenerate appends its anchor instead of replacing the message — pre-existing shape shared with Conversation) and #5927 (the game surface's sibling readers still see raw hidden rows).

9. **A live-delivered effect is lost, silently, in the five cases listed above.** The only trace it ever leaves is one warning, and only for the unwritable-reply case.
10. **The rendered reminder has no length budget of its own.** Every value it prints is a verbatim substring of a table file already refused above 64 KB, so the bound exists — but a maximal legal table would render roughly 60 KB into every Game turn's reminder. That is entirely self-inflicted by the package, and it is a named cost rather than a guarded one.

### What the tests prove, and what they do not

- The behavioral regression drives the executor against a real installed-package registry in a temporary data directory and a real chats store, because the claims worth pinning are claims about persisted state: the write lands with its ordinal stamped and `updatedAt` untouched, a second identical apply moves nothing, the live frame is emitted exactly once carrying the package id and the message triple, and every refusal tier (no table, no permission, oversized, tampered, malformed, not ready, wrong owner) is exercised without throwing. The byte ceiling is proved to key on the *declared* size: the same file on disk is refused when the manifest inflates it and accepted when the manifest tells the truth.
- **Three route behaviors are pinned by source-text search, not by execution** — the abort guard, the `metadata_patch` emit, and the once-per-turn resolution. The execution site lives inside a several-thousand-line route handler no regression can drive, and the file labels those three as searches rather than dressing them up as exercised behavior. Their *reachability* rests on review, not on a check.
- **The case-folding fix is not pinned**, and is not pinnable in process: it only misbehaves under a Turkish or Azeri runtime locale. The regression says so where it would otherwise be assumed.
- **Three render branches are not exercised by CI** — a number argument, a boolean argument, and an open string with no cap. All three were driven by hand and are correct; one extra fixture action would close the gap at near-zero cost.
- **No automated coverage of the prose experience.** Whether the rendered reminder actually gets a model to use these actions well is a playtest question, not a test question.

## Validation

- [ ] `pnpm check` passes locally
- [ ] Container (Docker / Podman) built and ran without issue
- [ ] Ran the app, clicked through the changes manually
- [ ] Checked edge cases (light + dark mode, mobile viewport, empty states, error paths)
- [ ] Above manual verification completed (describe below)
- [ ] Read and followed `CONTRIBUTING.md`

### Manual verification notes

**These twelve items need a Pixelforge build that ships a `gm-verbs.json` declaration installed as the chat's Experience.** No released package declares an action table, so on a stock install this PR is unobservable by design — a chat with no declaring package resolves zero actions, renders no extra prompt lines, parses nothing and writes nothing. Items 10–12 additionally need the package-side consumption of the standing action, which lives in the Agents repo.

- [ ] Manually verify in the browser that the GM emits a weather tag on a Pixelforge chat and that the tag is **not** visible in the saved narration.
- [ ] Manually read the chat's metadata row directly and confirm the value landed there — do this **before** looking at the sky, so a passing sky cannot mask a failed write.
- [ ] Manually verify in the browser that the sky changes without a reload.
- [ ] Manually verify that a deliberately malformed payload is still stripped out of the prose (the player must never see machinery), and that the world does not change.
- [ ] Manually verify in a Conversation-mode chat that none of this appears: no extra prompt lines, no parsing, no writes.
- [ ] Manually verify that streaming on a Game turn looks exactly as it did before.
- [ ] Manually verify that a Game chat with no Experience package is unchanged end to end.
- [ ] Manually verify that a **retried** turn still updates the sky (this is the retry-switch case that was previously missing).
- [ ] Manually verify that a **stopped** turn changes nothing — no write, no delivery, no claim.
- [ ] Manually verify in the browser that the GM emitting a standing action produces the package's own sentence and changes the world header standing word and the talk-window title, without a reload.
- [ ] Manually verify that regenerating that same turn does not stack anything — the write is absolute, and the last generated swipe wins.
- [ ] Manually verify that an unknown NPC name is refused by the package: no sentence, no change — and the tag was still stripped from the prose.

Alongside those:

- [ ] Run `pnpm check` and confirm it exits clean.
- [ ] Run the regressions: `capability-gm-verb-runtime`, `capability-gm-verbs`, `capability-conversation-actions`, `capability-package-lifecycle`, `prompt`.
- [ ] Confirm no `console.*` was introduced in server code (the shared logger is used throughout the new service).

During development `pnpm check` exited clean and those five regression lanes passed. That is context for the reviewer, not evidence for the boxes above — every browser item is genuinely unverified until someone runs it with a declaring package installed.

## Docs and release impact

- [ ] No docs changes needed
- [ ] Updated docs (README / CONTRIBUTING / android/README / CHANGELOG) as needed
- [ ] Translated docs on the `docs-i18n` branch updated to match, or a `[docs-i18n]` follow-up issue opened (see CONTRIBUTING.md § Translated documentation)
- [ ] Version/release files updated (only if this PR includes a version bump)

What this PR actually does, for the reviewer to confirm against the boxes:

- Adds the Capability API 1.16 section to `docs/development/optional-agent-packages.md` — the delivery-and-durability contrast, the mixed-turn ordering rule, the binding-versus-advisory refusal split, the honest gate scope, the fact that a table declared in the package's file list but left out of its asset paths yields zero actions and zero diagnostics, and the fact that a declared asset is served unguarded over HTTP so a table must never hold anything sensitive.
- Adds two CHANGELOG lines under Added (the second one is this PR's; the first belongs to the stacked schema PR and lands with it).
- Translated docs on `docs-i18n` are not updated here; this file's changes belong in the batched `[docs-i18n]` follow-up.
- No app version bump. The capability API constant moves from 1.15 to 1.16, which is a package-compatibility version and not one of the files `pnpm version:sync` owns.

## UI evidence (if applicable)

Nothing in this PR draws anything. The user-visible result is a world that changes when the narrator says it changed — which is a recording of a Pixelforge chat, not a screenshot of an Engine screen. Worth attaching if the browser pass above is recorded: one clip of the sky changing mid-turn without a reload.

## Summary by CodeRabbit

- **New Features**
  - Added support for package-declared Game Master actions in Game mode.
  - Actions can update chat state or be delivered live to connected packages when permitted.
  - Upgraded Capability API support to version 1.16.

- **Bug Fixes**
  - Game turns containing only hidden actions no longer erase the narration panel or get dropped.
  - Hidden, empty, and command-only turns are excluded from visible narration and navigation.

### CODE REVIEW

File: e2e/game-verb-only-turn.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            messagesPerPage: 20,
+            chatHelpSeenModes: ["conversation", "roleplay", "game"],
+            // Skip the typewriter so the assertion reads settled text.
+            gameInstantTextReveal: true,
+            // Off by default; the wheel-nav listener is a no-op without it.
+            gameMiddleMouseNav: true,
+          },
+          version: 99,
+        }),
+      );
```
[the remark this answers is not in the document]
Comment (kolacheee) [reply e2e/game-verb-only-turn.e2e.ts:99]: Fair point, declined here with a filed home: eight e2e files hand-write this exact payload today (core-flows alone at five sites), so this fixture follows the repo's shipped idiom, and adopting shared constants in one file would add a ninth variant without removing the drift risk. The repo-wide helper — one owner for the key, version, and a typed seed that could fail loudly on a renamed flag — is filed as #5928 with all eight files enumerated.
